eslint.config.js

```diff
@@ -0,0 +1,931 @@
+            }],
+        },
+    },
+    {
+        files: [
+            'ext/**/*.js',
+        ],
+
+        rules: {
+            'no-console': 'error',
+        },
+    },
+    {
+        files: [
+            'test/**/*.js',
+            'dev/**/*.js',
+            '**/integration.spec.js',
+            '**/playwright.config.js',
+            '**/playwright-util.js',
+            '**/visual.spec.js',
+        ],
+
+        languageOptions: {
+            globals: {
+                ...Object.fromEntries(Object.entries(globals.browser).map(([key]) => [key, 'off'])),
+                ...globals.node,
+                ...Object.fromEntries(Object.entries(globals.webextensions).map(([key]) => [key, 'off'])),
+            },
+        },
+    },
+    {
+        files: [
+            'test/data/html/**/*.js',
+        ],
+
+        languageOptions: {
+            globals: {
+                ...globals.browser,
+                ...Object.fromEntries(Object.entries(globals.node).map(([key]) => [key, 'off'])),
+                ...Object.fromEntries(Object.entries(globals.webextensions).map(([key]) => [key, 'off'])),
+            },
+
+            ecmaVersion: 5,
+            sourceType: 'script',
+        },
+    },
+    {
+        files: [
+            'test/data/html/**/*.js',
+        ],
+        ignores: [
+            'test/data/html/js/html-test-utilities.js',
+        ],
+
+        languageOptions: {
+            globals: {
+                HtmlTestUtilities: 'readonly',
+            },
+        },
+    },
+    {
+        files: [
+            'test/**/*.test.js',
+        ],
+        plugins: {
+            vitest,
+        },
+        ...vitest.configs.recommended,
+        rules: {
+            'vitest/prefer-to-be': 'off',
+        },
+    },
+    ...compat.extends('plugin:@typescript-eslint/disable-type-checked').map((config) => ({
+        ...config,
+        files: [
+            'dev/lib/**/*.js',
+            'eslint.config.js',
+        ],
+    })),
+    {
+        files: [
+            'ext/js/templates/template-renderer-frame-main.js',
+            ...await getDependencies(['ext/js/templates/template-renderer-frame-main.js']),
```

Comment:
Lines like these are what allows us to safely get rid of `eslint-config.test.js`.